src/peer_digest.cc

```diff
@@ -134,40 +129,20 @@ peerDigestCreate(CachePeer * p)
 
     // TODO: make CachePeer member a CbcPointer
     p->digest = cbdataReference(pd);
-
-    // lock a reference to pd again to prevent the PeerDigest
-    // disappearing during peerDigestDestroy() when
-    // cbdataReferenceValidDone is called.
-    // TODO test if it can be moved into peerDigestDestroy() or
-    //      if things can break earlier (eg CachePeer death).
-    (void)cbdataReference(pd);
 }
 
 /* call Clean and free/unlock everything */
 static void
 peerDigestDestroy(PeerDigest * pd)
 {
-    void *p;
     assert(pd);
-    void * peerTmp = pd->peer;
-
-    /*
-     * DPW 2007-04-12
-     * We locked the peer in PeerDigest constructor, this is
-     * where we unlock it.
-     */
-    if (cbdataReferenceValidDone(peerTmp, &p)) {
-        // we locked the p->digest in peerDigestCreate()
-        // this is where we unlock that
-        cbdataReferenceDone(static_cast<CachePeer *>(p)->digest);
-    }
-
     delete pd;
 }
 
 PeerDigest::~PeerDigest()
 {
     delete cd;
+    cbdataReferenceDone(peer);
```

Comment:
Do you think we should convert PeerDigest::peer to CbcPointer in _this_ PR? I think the bug fix is easier to understand if that conversion is postponed to a dedicated commit, but I am not sure. Perhaps we should implement that conversion as a separate branch commit and then revert it for the official PR (while promising to post that reverted commit as a separate followup PR)?

Reply:
Looks like a good idea, converted at 9c791262. 

>  promising to post that reverted commit as a separate followup PR

I would do thus to minimize this PR changes.